test/unit/cmap/handshake/default_handshake_decorator.test.ts

```diff
@@ -0,0 +1,98 @@
+import { expect } from 'chai';
+import * as os from 'os';
+import * as sinon from 'sinon';
+
+import {
+  Connection,
+  DefaultHandshakeDecorator,
+  HostAddress,
+  LEGACY_HELLO_COMMAND
+} from '../../../mongodb';
+
+describe('DefaultHandshakeDecorator', function () {
+  const metadata = {
+    driver: {
+      name: 'Node',
+      version: '5.0.0'
+    },
+    os: {
+      type: os.type(),
+      name: process.platform,
+      architecture: process.arch,
+      version: os.release()
+    },
+    platform: 'MacOS'
+  };
+  const options = {
+    id: 1,
+    generation: 2,
+    hostAddress: new HostAddress('127.0.0.1:27017'),
+    monitorCommands: false,
+    tls: false,
+    loadBalanced: false,
+    metadata: metadata
+  };
+
+  describe('#generate', function () {
+    context('when the auth context exists', function () {
```

Comment:
We identified we still have a bug, from the server logs:

```json
{"t":{"$date":"2023-03-28T17:15:09.807-04:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn2","msg":"client metadata","attr":{"remote":"127.0.0.1:65369","client":"conn2","doc":{"driver":{"name":"nodejs","version":"5.1.0"},"os":{"type":"Darwin","name":"darwin","architecture":"arm64","version":"21.6.0"},"platform":"Node.js v18.12.1, LE (unified)|Node.js v18.12.1, LE (unified)"}}}
```

They are tricky to setup but should we strive to add some integration tests where we use the driver from highlevel API only (runCommand, ping:1) and we use spys / FakeSocket to validate handshakes?